modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeClientIndex.java

```diff
@@ -75,12 +75,20 @@ public static H2TreeClientIndex createIndex(
         IndexType idxType = pk ? IndexType.createPrimaryKey(false, false) :
             IndexType.createNonUnique(false, false, false);
 
-        CacheConfiguration ccfg = tbl.cacheInfo().config();
+        GridCacheContextInfo cacheInfo = tbl.cacheInfo();
+
+        CacheConfiguration ccfg = cacheInfo.config();
 
         List<InlineIndexColumn> inlineCols = getAvailableInlineColumns(false, ccfg.getName(),
-            idxName, log, pk, tbl, cols, new InlineIndexColumnFactory(tbl.getCompareMode()), true);
+                idxName, log, pk, tbl, cols, new InlineIndexColumnFactory(tbl.getCompareMode()), true);
+
+        int maxAllowedInlineSize = H2TreeIndexBase.maxAllowedInlineSize(
```

Comment:
It is incorrect to calculate value on client, because page size is unknown, EncryptionSpi can? be unavailable on client.
Maybe better always return -1 ?